README.md

```diff
@@ -68,26 +68,39 @@ Name of the remote branch to link to.
 
 ### Git Timemachine
 
-If [`git-timemachine-mode`](https://github.com/pidu/git-timemachine) is active `git-link` generates a URL for the version of the file being visited.
+If [`git-timemachine-mode`](https://github.com/pidu/git-timemachine)
+is active `git-link` generates a URL for the version of the file being
+visited.
 
 ### Building Links and Adding Services
 
-`git-link-remote-alist` and `git-link-commit-remote-alist` map remotes'
-hostnames to a function capable of creating a URL on that host. To add (or
-modify) how URLs are created for a given host add the appropriate function
-objects to these lists.
+`git-link-remote-alist` is an alist containing `(REGEXP FUNCTION)`
+elements. The FUNCTION creates URLs for file on remote host names that
+match the REGEXP. To add (or modify) how URLs are created for a given
+host, add appropriate elements to this list.
 
-If you use a self-hosted version of one of the supported services, you
-can configure the link function alists for the hostname at which that
-service is hosted.  For example, for a GitHub Enterprise instance at
-`github.example.com`, you could add the following to your `.emacs` file.
+As an example, one of the default elements in this alist is
+`("gitlab" git-link-gitlab)`. So the `git-link-gitlab` function
+will be used to create URLs to files in remotes that match the
+*regexp* `"gitlab"`.  That would cover common Gitlab host URLs like
+*"gitlab.com"*, *"gitlab.example.com"* and *"gitlab.example.org"*.
+
+`git-link-commit-remote-alist` is also an alist containing `(REGEXP
+FUNCTION)` elements. Here, the FUNCTION creates URLs to the commit
+pages, for remote hosts matching REGEXP.
+
```

Comment:
I have some minor nitpicks here but I don't want to beat you over the head too much about this after all your work. I will update later.